### PR TITLE
test: add tests for sources

### PR DESCRIPTION
**Pull request by zarov**
Add more tests for sources, and add the sinon dependency to test the console output (to cover more branch, but I'm not convinced by the utility of this).

### CODE REVIEW

File: test/unit/source.js [hunk 1]
```diff
@@ -30,104 +31,260 @@ global.URL = function URL() {
...
-
-describe('Source', function () {
-    it('Should instance and throw error for Source', function () {
-        const source = new Source({ url: 'http://'  });
-        assert.throws(source.urlFromExtent, Error);
-        assert.throws(source.extentInsideLimit, Error);
-        assert.throws(source.extentsInsideLimit, Error);
+describe('Sources', function () {
+    beforeEach(function () {
+        sinon.spy(console, 'warn');
```
Comment (gchoqueux) [inline test/unit/source.js]: The test fails if `sinon` finds a warn console?

File: package.json [hunk 2]
```diff
@@ -80,6 +80,7 @@
     "proj4": "^2.5.0",
     "puppeteer": "^1.14.0",
     "replace-in-file": "^3.4.4",
+    "sinon": "^7.3.1",
```
Comment (gchoqueux) [inline package.json]: * You use Sinon to spy Console, What do you test exactly?

* I don't know if [startStaticFileServer](https://github.com/iTowns/itowns/blob/master/test/functional/bootstrap.js#L22) is fake server? if yes we could use [sinon fake server](https://sinonjs.org/releases/v7.3.1/fake-xhr-and-server/#fake-server).

* Sinon could be used to make mocks, too.

**Review by gchoqueux — COMMENTED** [review]
It's good to add tests.

File: test/unit/source.js (hunk 1 above)
Comment (zarov) [reply test/unit/source.js]: Before each test, `sinon` is spying on what we tell it: here it is `console.warn`. It modifies the `console.warn` method, and allow us to do some tests later on (see below `console.warn.calledOnce` for example).

File: package.json (hunk 2 above)
Comment (zarov) [reply package.json]: Yes, `sinon` seems to be quite complete. I'm looking more into it then.

Comment (zarov) [reply package.json]: I'm dropping `sinon` and letting the PR as it is without it.

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**